test/mciproject.yml

```diff
@@ -58,69 +83,86 @@ tasks:
       - command: git.apply_patch
         params:
           directory: wiredtiger
+      - func: "compile wiredtiger"
+
+  - name: make-check-test
+    depends_on:
+      - name: compile
+    commands:
+      - func: "fetch artifacts"
+      - func: "compile wiredtiger"
       - command: shell.exec
         params:
           working_dir: "wiredtiger"
           script: |
             set -o errexit
             set -o verbose
-            if [ "Windows_NT" = "$OS" ]; then
-              scons.bat --enable-python=c:\\swigwin-3.0.2\\swig.exe --enable-diagnostic ${smp_command|}
-            else
-              ./build_posix/reconf
-              ${configure_env_vars|} ./configure --enable-diagnostic --enable-python --enable-zlib --enable-strict
-              ${make_command|make} ${smp_command|} 2>&1
-
-              # On macOS, change the binary location with install_name_tool since DYLD_LIBRARY_PATH
-              # appears not to work for dynamic modules loaded by python. For wt, the libtool generated
-              # script has the wrong path for running on test machines.
-              if [ "$(uname -s)" == "Darwin" ]; then
-                WT_VERSION=$(m4 build_posix/aclocal/version.m4)
-                install_name_tool -change /usr/local/lib/libwiredtiger-$WT_VERSION.dylib $(pwd)/.libs/libwiredtiger-$WT_VERSION.dylib lang/python/_wiredtiger*.so
-                install_name_tool -change /usr/local/lib/libwiredtiger-$WT_VERSION.dylib $(pwd)/.libs/libwiredtiger-$WT_VERSION.dylib .libs/wt
-              fi
-
-              ${test_env_vars|} TESTUTIL_ENABLE_LONG_TESTS=1 ${make_command|make} VERBOSE=1 check 2>&1
-            fi
-      - command: archive.targz_pack
+
+            ${test_env_vars|} TESTUTIL_ENABLE_LONG_TESTS=1 ${make_command|make} VERBOSE=1 check ${smp_command|} 2>&1
+
+  - name: examples-c-test
+    depends_on:
+      - name: compile
+    commands:
+      - func: "fetch artifacts"
+      - func: "compile wiredtiger"
+      - command: shell.exec
         params:
-          target: "wiredtiger.tgz"
-          source_dir: "wiredtiger"
-          include:
-             - "./**"
-      - command: s3.put
+          working_dir: "wiredtiger"
+          script: |
+            set -o errexit
+            set -o verbose
+
+            ${test_env_vars|} ${make_command|make} VERBOSE=1 check -C examples/c  ${smp_command|} 2>&1
+
+  - name: csuite-col-append-test
```

Comment:
I'd prefer it if this name were auto-generateable from either the directory structure or the test name - so we can easily script the creation of this file at some stage.

Reply:
For now I manipulate the test name a bit to get the task name in a manual approach - e.g. for test name `test_wt2246_col_append`, remove the `test_` prefix and `wt2246_` ticket info, then  add `csuite-` prefix and `-test` postfix to get the task name `csuite-col-append-test`. Should be okay to scriptize it later.

Reply:
How to figure out if a test supports `TESTUTIL_ENABLE_LONG_TESTS` is an open question. Searching the keyword in source code feels less than ideal.